src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -43,39 +48,67 @@ public void createCourse(Course course) throws InterruptedException, ExecutionEx
      * @return An ArrayList of Course objects associated with the specified instructor.
      * @throws RuntimeException If an InterruptedException or ExecutionException occurs during Firestore operations.
      */
-    public ArrayList<Course> getCoursesByInstructor(Instructor instructor) throws InterruptedException, ExecutionException {
+    public ArrayList<Course> getCoursesByInstructor(Instructor instructor) {
```

Comment:
Let's change this to just `instructorID` being passed in instead of instructor object

Reply:
Also, if the provided instructorID is null, throw `NullPointerException`, no need to catch or indicate that func throws since it extends `RuntimeException`